### PR TITLE
Fix Sanity Tests

### PR DESCRIPTION
**Pull request by gomesjason**
**Is this a bug fix or adding new feature?**
Fix sanity tests

**What is this PR about? / Why do we need it?**
This PR fixes the sanity tests in a few ways. It fixes failures that were occurring, tests both filesystem and volume cases,  and updates the dependency version from v1 to v5.

**What testing is done?** 
Sanity testing

### CODE REVIEW

**Comment by jacobwolfaws** [comment]
@gomesjason can you revise the commit to break out the sanity testing change + the bug fix along with un-bumping the csi driver image version. That's part of the release process and shouldn't be included here

**Comment by gomesjason** [comment]
> @gomesjason can you revise the commit to break out the sanity testing change + the bug fix along with un-bumping the csi driver image version. That's part of the release process and shouldn't be included here

Fixed

File: pkg/cloud/cloud.go [hunk 1]
```diff
@@ -169,6 +169,7 @@ func (c *cloud) ResizeFileSystem(ctx context.Context, fileSystemId string, newSi
 				if err != nil {
 					return nil, err
 				}
+				return &newSizeGiB, nil
```
Comment (jacobwolfaws) [inline pkg/cloud/cloud.go:172]: Note, @hughdanliu and I discussed making this a separate commit/PR offline. Is this causing the sanity tests to fail @gomesjason? If not, I'm inclined to believe this needs an additional sanity test to catch it, if so, can you break this into a separate commit for cleanliness sake

Comment (jacobwolfaws) [reply pkg/cloud/cloud.go:172]: correction, @hughdanliu pointed out that the sanity tests wouldn't be expected to cover this. Would it be possible to break it into its own commit and add a corresponding unit test

Comment (gomesjason) [reply pkg/cloud/cloud.go:172]: Yea, added it since it breaks a unit test but we can move it elsewhere

Comment (hughdanliu) [reply pkg/cloud/cloud.go:172]: There was an existing unit test to address this situation but it was broken on release. Just for the sake of velocity this particular bug fix seems fine to leave it in here, but if @jacobwolfaws feels otherwise I don't feel strongly about this.

Comment (jacobwolfaws) [reply pkg/cloud/cloud.go:172]: I'd prefer it if you staggered the commits, the first being the bug fix and the second being the sanity tests. It's cleaner that way imo

Comment (gomesjason) [reply pkg/cloud/cloud.go]: Fixed

File: tests/sanity/sanity_test.go [hunk 2]
```diff
@@ -55,16 +54,53 @@ var _ = AfterSuite(func() {
...
+		}
+
+		BeforeAll(func() {
+			fsxOpenZfsDriver.ResetCloud()
+		})
+
+		Describe("CSI sanity", func() {
+			BeforeEach(func() {
+				switch {
+				case strings.Contains(CurrentSpecReport().FullText(), "should create volume from an existing source snapshot"),
```
Comment (hughdanliu) [inline tests/sanity/sanity_test.go:72]: Discussed offline - the tests that we're excluding here are some of the tests that were newly added as part of `github.com/kubernetes-csi/csi-test/v5/pkg/sanity`. Some are not applicable to our implementation (e.g. snapshots of file systems), while others should be caught at the Kubernetes/sidecar layer. Adding another layer to protect against these situations here would require a large effort and involve making additional API calls. I think these are fine to exclude.

**Comment by gomesjason** [comment]
Fixing an unnecessary skip

**Comment by gomesjason** [comment]
Fixed, added comments to justify skipped tests
